src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +395,422 @@ else if (!period.isEmpty()) {
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummary = getInlineSummary(ast);
+        if (inlineSummary.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final boolean isPeriodAbsent = isPeriodAtEnd(inlineSummary, period);
+            if (isPeriodAbsent) {
+                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+            }
+            else if (isSummaryValid(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Get summary sentence for inline type summary javadoc.
+     *
+     * @param javadoc javadoc root node.
+     * @return Summary sentence of javadoc root node.
+     */
+    private static String getInlineSummary(DetailNode javadoc) {
+        final String summary;
+        final DetailNode descriptionNode = getDescriptionNode(javadoc);
+        final boolean isSummaryTagWithinHtml = isSummaryTagWithinHtmlTag(javadoc);
+        if (isSummaryTagWithinHtml) {
+            summary = getInlineHtmlSentence(javadoc);
+        }
+        else if (descriptionNode == null) {
+            summary = "";
+
+        }
+        else {
+            final String summarySentence = getDefaultInlineSummarySentence(descriptionNode);
+            summary = removeHtmlTagsFromString(summarySentence);
+        }
+        return summary;
+    }
+
+    /**
+     * Checks if summary tag present as HTML format.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean isSummaryTagWithinHtmlTag(DetailNode javadoc) {
+        boolean found = false;
+        final DetailNode summaryNode = getInlineTagNodeInsideHtmlTagsOnly(javadoc);
+        if (summaryNode != null) {
+            found = isSummaryTag(summaryNode);
+        }
+        return found;
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Checks if the summary is valid.
+     *
+     * @param ast Javadoc root node.
+     * @return true, if sentence is invalid.
+     */
+    private boolean isSummaryValid(DetailNode ast) {
+        final String inlineFirstSentence = getInlineSummary(ast);
+        final int endOfInlineSentence = inlineFirstSentence.lastIndexOf(period);
+        final String summary = inlineFirstSentence.substring(0, endOfInlineSentence);
+        return containsForbiddenFragment(summary)
+                || !containsCorrectInlineTags(ast);
```

Comment:
I do not think this method is required, after our latest agreement. 
if you still think it matter, please show me in Inputs a case.

Reply:
It was there to ensure violation for cases like 
```
  /**
   * {@summary {@code see} .} // no summary
   */
  void foo10() {} // ok
```

But since we are changing the check now, this is no more required now.

Reply:
This should not be violation.

Reply:
> This should not be violation. We do not care about content of summary, if it not empty it is ok, and it should finish with period.

This is the current behaviour, there are also several test cases already included. 

https://github.com/checkstyle/checkstyle/blob/15edaeb8196e06ddc875710b169e69673367ea27/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineDefault.java#L178-L181

Reply:
please generate HTML for this method and share here image of summary row for this method.

Reply:
![Screenshot from 2021-04-13 19-36-15](https://user-images.githubusercontent.com/69365878/114566172-a48ea400-9c8f-11eb-9d71-7c295c1bf034.png)

This is no more a violation.